### PR TITLE
Add centos dockerfile

### PR DESCRIPTION
**Pull request by yixinglu**
Compile dependencies with gcc in devtoolset-8

Refer to vesoft-inc/nebula#838
close vesoft-inc/nebula#905

### CODE REVIEW

File: Dockerfile.centos [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    maven \
+    ncurses-devel \
+    perl \
+    perl-WWW-Curl \
+    readline-devel \
+    rpm-build \
+    unzip \
+    vim \
+    wget \
+    xz-devel \
```
Comment (laura-ding) [inline Dockerfile.centos]: install gdb

Comment (darionyaphet) [reply Dockerfile.centos]: Does `curl` is needed ？

Comment (yixinglu) [reply Dockerfile.centos]: `gdb` has been installed by `devtoolset-8` package and `curl` also has been installed by other dependencies.

File: Dockerfile.centos [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  && yum clean all \
+  && rm -rf /var/cache/yum
+
```
Comment (dutor) [inline Dockerfile.centos]: Could you please add some notes for each primary step?

File: Dockerfile.centos [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+RUN scl enable devtoolset-8 -- bash \
+  && mkdir -p /home/nebula \
+  && ln -snf /opt/rh/devtoolset-8/enable /etc/profile.d/devtoolset-8.sh
```
Comment (dutor) [inline Dockerfile.centos]: How about changing the target name to `devtoolset-8-setup.sh` or other meaningful names?

**Pull request merged**